subprojects/snapshots/src/main/java/org/gradle/internal/vfs/impl/VfsRootReference.java

```diff
@@ -14,34 +14,32 @@
  * limitations under the License.
  */
 
-package org.gradle.internal.snapshot;
+package org.gradle.internal.vfs.impl;
+
+import org.gradle.internal.snapshot.SnapshotHierarchy;
 
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.UnaryOperator;
 
-public class AtomicSnapshotHierarchyReference {
+public class VfsRootReference {
```

Comment:
Have you considered making this into an `AbstractVirtualFileSystem` instead?

Reply:
I prefer delegation in this place.